src/util.ts

```diff
@@ -1228,3 +1244,26 @@ export function getStandaloneEvals(): StandaloneEval[] {
   });
   return flatResults;
 }
+
+export function providerToIdentifier(provider: TestCase['provider']): string | undefined {
```

Comment:
Really appreciate you adding these helper functions.  I'm aware that ApiProvider vs ProviderOptions and similar variations are some of the ugliest parts of the code :(

Reply:
No prob. I love the flexibility of `promptfoo`. You can use it many ways. But it also does brings on complexity. I suspect little utility functions to simplify logic can go a long way.